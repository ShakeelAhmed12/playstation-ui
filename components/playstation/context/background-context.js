import { createContext, useState } from 'react';

export const BackgroundContext = createContext();

export const BackgroundProvider = ({ children }) => {
    const [activeGame, setActiveGame] = useState('explore');

    return (
        <BackgroundContext.Provider value={{ activeGame, setActiveGame }}>
            {children}
        </BackgroundContext.Provider>
    )
}