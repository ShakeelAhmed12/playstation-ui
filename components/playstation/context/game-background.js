import { createContext, useState } from 'react';

export const BackgroundContext = createContext();

export const GameBackground = ({ children }) => {
    const [activeGame, setActiveGame] = useState('explore');

    return (
        <BackgroundContext.Provider value={{ activeGame, setActiveGame }}>
            {children}
        </BackgroundContext.Provider>
    )
}