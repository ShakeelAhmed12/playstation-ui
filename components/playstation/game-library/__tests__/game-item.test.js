import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { GameBackground } from '../../context/game-background';

import GameItem from '../component/game-item';

let realUseContext;
let useContextMock;
// Setup mock
beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
});
// Cleanup mock
afterEach(() => {
    React.useContext = realUseContext;
});

describe('Game Item tests', () => {
    it('Game item renders correctly', () => {
        const tree = renderer
            .create(
                <GameBackground>
                    <GameItem gameImg="/playstation/wallpapers/explore-bg.jpg" title="News" />
                </GameBackground>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})