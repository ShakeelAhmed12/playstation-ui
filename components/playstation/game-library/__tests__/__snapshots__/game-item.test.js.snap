// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Item tests Game item renders correctly 1`] = `
<div
  className="game-item__GameImage-sc-11p7q2j-0 liplco"
  id="background-wrapper"
  onClick={[Function]}
/>
`;
